### PR TITLE
feat: IS-3260: Staleness check workflow step

### PR DESCRIPTION
**Pull request by dlongeuay**
This PR implements some of the pre-work needed for https://github.com/iStreamPlanet/kubernetes_management/issues/3672

## Notes:
- Added new helmfile-action `staleness`
- Default for staleness is 30 days

This PR corresponds with https://github.com/iStreamPlanet/kubernetes_management/pull/3868

### CODE REVIEW

**Review by CharlieSu — APPROVED** [review]

**Review by marcind — CHANGES_REQUESTED** [review]
I don't think adding this to `helmfile-actions` is the right thing to do. The purpose of `helmfile_actions` is to provide a wrapper around the `helmfile` sub-commands, not to be a kitchen sink of everything related to helmfile. This functionality does not seem to even call `helmfile`, which to me indicates that it does not fit in this bucket. I know it will take a bit more work, but I think this functionality should be in a separate folder, something like `helmfile-lock-stale`.

File: helmfile-actions/staleness.sh [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  if [ ${isMissing} = "false" ]; then
+    dateGenerated=`cat helmfile.lock | sed -n -r "s/^generated:\s\"(.*)\"/\1/gp"`
+    dateGeneratedSinceEpoch=$(date -d ${dateGenerated} +%s)
+    dateCurrentSinceEpoch=$(date +%s)
+    dateDeltaSeconds=$((${dateCurrentSinceEpoch} - ${dateGeneratedSinceEpoch}))
+    dateDeltaDays=$((${dateDeltaSeconds} / (24*3600)))
+
+    staleCode=2
+    if [ ${dateDeltaDays} -le ${daysStale} ]; then
```
Comment (marcind) [inline helmfile-actions/staleness.sh:27]: I'd have to look up how bash handles integer division, but comparing the staleness in terms of seconds will be more precise, as it lets you avoid issues with integer division losing fractional days precision.

File: helmfile-actions/main.sh [hunk 2]
```diff
@@ -1,16 +1,18 @@
 #!/bin/bash
 
-if [ "$#" -ne 2 ]; then
-  echo 'Usage: ./main.sh $command $working_directory'
+if [ "$#" -ne 3 ]; then
+  echo 'Usage: ./main.sh $command $working_directory $days_stale'
```
Comment (marcind) [inline helmfile-actions/main.sh:4]: This is another symptom that you are co-opting a generic command wrapper into something specific to this staleness scenario by reserving a positional parameter. Also, making this change will break existing other usages of this action.

Comment (dlongeuay) [reply helmfile-actions/main.sh]: good call, I moved this to a separate action

File: helmfile-actions/staleness.sh (hunk 1 above)
Comment (dlongeuay) [reply helmfile-actions/staleness.sh]: fixed, the check is now based on seconds. I am still returning an approximate delta in days for use in the issue that will be created. I am not too familiar with floating point division in bash and not sure if it is even worth the headache to return a decimal.

**Review by marcind — APPROVED** [review]

**Review by CharlieSu — APPROVED** [review]

**Pull request merged**